app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -151,9 +152,20 @@ class StateFragmentPresenter @Inject constructor(
     moveToNextState()
   }
 
-  fun onNextButtonClicked() = moveToNextState()
+  fun onNextButtonClicked() {
+    val item = hintBulbVibisilityStack.lastOrNull()
```

Comment:
Please add tests to verify the new functionality is correct (specifically that the test fails without the fix & passes with it).

Reply:
Done